hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -711,11 +716,31 @@ private void dispatchSyntheticTxn(
             childStack.commitFullStack();
         } catch (final HandleException e) {
             childRecordBuilder.status(e.getStatus());
-            recordListBuilder.revertChildrenOf(recordBuilder);
+            if (childCategory == SCHEDULED) {
+                if (e.shouldRollbackStack()) {
+                    childStack.rollbackFullStack();
+                }
+                final var userTransactionRecordBuilder = recordListBuilder.userTransactionRecordBuilder();
+                userTransactionRecordBuilder.status(e.getStatus());
+                recordListBuilder.revertChildrenOf(userTransactionRecordBuilder);
```

Comment:
Does this remove the child record (seems like it would) or write the child record with failure status?
I believe the rollbackFullStack above removes the state changes, but I think we want to _keep_ the child record(s) in this case.

Reply:
Very confusing method/variable names (it seems it would be writing the parent and reverting the parent's children)...